graphbolt/include/utils.h

```diff
@@ -0,0 +1,151 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file graphbolt/include/csc_sampling_graph.h
+ * @brief Header file of csc sampling graph.
+ */
+
+#include <torch/torch.h>
+
+#include <string>
+#include <vector>
+
+namespace graphbolt {
+namespace utils {
+
+/**
+ * @brief Utility function to write to archive.
+ * @param archive Output archive.
+ * @param key Key name used in saving.
+ * @param data Data that could be constructed as `torch::IValue`.
+*/
+template <typename DataT>
+void write_to_archive(
+    torch::serialize::OutputArchive& archive,
+    const std::string& key,
+    const DataT& data) {
+  archive.write(key, data);
```

Comment:
Should we put all implementations in .cc file?

Reply:
It's template functions,  I think it's ok to be header only.

Reply:
But there is also template instantiation e.g. `void write_to_archive<std::vector<std::string>>(`.

If you have to put it in header.

Add
`
#ifndef GRAPHBOLT_INCLUDE_UTILS_H_
#define GRAPHBOLT_INCLUDE_UTILS_H_
...
#endif
`

example: https://source.chromium.org/chromium/chromium/src/+/main:base/check_deref.h;drc=cda36986fdb5e85aeeaa6e106e77fea17574ad4a;l=12